dandi/download.py

```diff
@@ -201,17 +201,29 @@ def download_generator(
 
             downloader = client.get_download_file_iter(*down_args)
 
+            if isinstance(client, girder.GirderCli):
+                mtime = asset.get("modified")
+            else:
+                assert isinstance(client, DandiAPIClient)
+                try:
+                    mtime = asset["metadata"]["blobDateModified"]
+                except KeyError:
+                    mtime = None
+                if mtime is None:
+                    mtime = asset.get("modified")
```

Comment:
not 100% sure we should do that (set to mtime of the asset record itself) but leaving it without any mtime would not be better... 
I think we should keep the logic, but also log a warning here describing the situation since (after we are done with this) all assets should have blobDateModified (if uploaded by our client at least... may be we should even make that field non-optional? is there a use case where it could be not provided?). WDYT @jwodder ?

Reply:
Warning added.